coderd/userauth.go

```diff
@@ -1531,7 +1537,9 @@ func (api *API) oauthLogin(r *http.Request, params *oauthLoginParams) ([]*http.C
 			ignored := make([]string, 0)
 			filtered := make([]string, 0, len(params.Roles))
 			for _, role := range params.Roles {
-				if _, err := rbac.RoleByName(role); err == nil {
+				// TODO: This only supports mapping deployment wide roles. Organization scoped roles
+				// are unsupported.
+				if _, err := rbac.RoleByName(rbac.RoleIdentifier{Name: role}); err == nil {
```

Comment:
follow-up for TODO

also, should we catch an attempt to lookup an org-scoped role here?

Reply:
Followup is already an issue: https://github.com/coder/coder/issues/12147

And we do check that an org scoped role is not assigned here:

https://github.com/coder/coder/blob/9bf3298217faef43da269fedb04c72795b93325d/coderd/rbac/roles.go#L584-L584

If you pass a nil uuid, and some org permissions exist, it will fail.